src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/JsonDescriptor.php

```diff
@@ -399,14 +399,14 @@ private function describeValue($value, $omitTags, $showArguments)
             );
         }
 
-        if ($value instanceof Definition) {
-            return $this->getContainerDefinitionData($value, $omitTags, $showArguments);
-        }
-
         if ($value instanceof ArgumentInterface) {
             return $this->describeValue($value->getValues(), $omitTags, $showArguments);
         }
 
+        if ($value instanceof Definition) {
```

Comment:
The existing code won't get a new ServiceLocatorArgument all of a sudden, so if they do, someone did write some new code - thus we're out of BC concerns.
Processing ServiceLocatorArgument as a Definition wouldn't be a big issue btw. Existing code won't get the added semantic, but they will not break. Note that this is also true for pure ArgumentInterface instances also.